### DISCUSSION TITLE
Got "Uncaught ReferenceError: grecaptcha is not defined" when open the custom recaptcha form page

### DISCUSSION THREAD

**Discussion by Mizogorou** [breezefront/community#206] [1. Help] [answered]
Hello,

I have created a custom contact form with recaptcha version 3 on https://www.tronictoy.com/payment-confirmation/ but I am unable to submit it successfully the first time due to the error "Uncaught ReferenceError: grecaptcha is not Defined"

So I had to refresh the page first and the error disappeared and I was able to successfully submit the contact form.

This error is never displayed in the Luma theme on https://test.tronictoy.com/payment-confirmation/

How can I fix this problem?

Thank you.

**Comment by vovayatsyuk** [comment]
I'm not sure how to reproduce the error. When I open https://www.tronictoy.com/payment-confirmation I see that `grecaptcha` variable is available in the console.

**Comment by Mizogorou** [comment]
Please scroll to the footer section and click on 2nd line of menu to open link. This method works for me.

<img width="981" alt="Screenshot 2566-10-30 at 20 45 09" src="https://github.com/breezefront/community/assets/8059336/ec39bd60-d838-4704-9ff7-a6b78a7ce841">

**Reply by vovayatsyuk** [reply]
When I do this, I see the error from the easybanner module. This error was fixed since easybanner 1.9.7. Please update the module.

**Comment by Mizogorou** [comment]
Thank you for your update about easybanner. However, "Uncaught ReferenceError: grecaptcha is not Defined" error still persist. 

Please see my video for more detail. I have uploaded to my hosting because this file larger than 10 MB.
https://test.tronictoy.com/recaptcha-error.mov

**Reply by vovayatsyuk** [reply]
Thanks, I was able to reproduce this error. 

Here is the code that cause the error:

```js
<script>
  grecaptcha.ready(function() {
    grecaptcha.execute('...', {action: 'validate_captcha'}).then(function(token) {
       // console.log(token);
       document.getElementById("g-recaptcha-response").value = token;
    });
    // refresh token every minute to prevent expiration
    setInterval(function(){
      grecaptcha.execute('...', {action: 'validate_captcha'}).then(function(token) {
        console.log( 'refreshed token:', token );
        document.getElementById("g-recaptcha-response").value = token;
      });
    }, 60000);

  });
</script>
```

This code doesn't guarantee that the grecaptcha variable exists. You need to rewrite it in a way that will ensure that grecaptcha variable is created before this code is executed.

**Reply by Mizogorou** [reply]
Thank you for your information.

However, Why does this problem is not appear on Magento Luma?

**Reply by vovayatsyuk** [reply, marked as answer]
If I'm correct, when Turbo is enabled, scripts inside the body are handled asynchronously. You can disable Turbo mode in Breeze settings: https://breezefront.com/docs/settings

Turbolinks docs: https://github.com/turbolinks/turbolinks#working-with-script-elements

**Reply by Mizogorou** [reply]
Thank you vovayatsyuk,

You're right. It worked.
